raiden/blockchain_events_handler.py

```diff
@@ -32,44 +31,43 @@
 def handle_tokennetwork_new(raiden, event: Event):
     """ Handles a `TokenNetworkCreated` event. """
     data = event.event_data
-    token_network_address = data['token_network_address']
+    args = data['args']
+    block_number = data['block_number']
+    token_network_address = args['token_network_address']
+    token_address = typing.TokenAddress(args['token_address'])
 
     token_network_proxy = raiden.chain.token_network(token_network_address)
     raiden.blockchain_events.add_token_network_listener(
         token_network_proxy=token_network_proxy,
         contract_manager=raiden.contract_manager,
-        from_block=data['blockNumber'],
-    )
-
-    token_address = typing.TokenAddress(
-        data_decoder(event.event_data['args']['token_address']),
+        from_block=block_number,
     )
 
     token_network_state = TokenNetworkState(
         token_network_address,
         token_address,
     )
 
-    transaction_hash = event.event_data['transactionHash']
-    assert transaction_hash, 'A mined transaction must have the hash field'
+    transaction_hash = event.event_data['transaction_hash']
 
     new_token_network = ContractReceiveNewTokenNetwork(
         transaction_hash=transaction_hash,
         payment_network_identifier=event.originating_contract,
         token_network=token_network_state,
-        block_number=data['block_number'],
+        block_number=block_number,
     )
     raiden.handle_state_change(new_token_network)
 
 
 def handle_channel_new(raiden, event: Event):
     data = event.event_data
+    block_number = data['block_number']
+    args = data['args']
     token_network_identifier = event.originating_contract
-    transaction_hash = event.event_data['transactionHash']
-    assert transaction_hash, 'A mined transaction must have the hash field'
```

Comment:
This is one example of the `transaction_hash` check, this is not in the `decode_event_to_internal` and we don't have to care about it in these handlers